docs/source/advanced.rst

```diff
@@ -178,11 +178,50 @@ This is basically the same as calling :py:func:`~scikeras.wrappers.BaseWrapper.g
 Data Transformers
 ^^^^^^^^^^^^^^^^^
 
-In some cases, the input actually consists of multiple inputs. E.g.,
+Keras supports a much wider range of inputs/outputs than Scikit-Learn does. E.g.,
 in a text classification task, you might have an array that contains
 the integers representing the tokens for each sample, and another
-array containing the number of tokens of each sample. SciKeras has you
-covered here as well.
+array containing the number of tokens of each sample.
+
+In order to reconcile Keras' expanded input/output support and Scikit-Learn's more
+limited options, SciKeras introduces "data transformers". These are really just
+dependency injection points where you can declare custom data transformations,
+for example to split an array into a list of arrays, join `X` & `y` into a `Dataset`, etc.
+In order to keep these transformations in a familiar format, they are implemented as
+sklearn-style transformers. You can think of this setup as an sklearn Pipeline:
+
+.. code-block::
+
+                                             ↗ feature_encoder ↘
+    your data → sklearn-ecosystem → SciKeras                    dataset_transformer → Keras
```

Comment:
I think this diagram is really useful.

Would this be a better diagram?


```
                                   ↗ feature_encoder ↘
    SciKeras.fit(features, labels)                    dataset_transformer → Keras.fit(dataset)
                                   ↘ target_encoder  ↗ 
```

Reply:
I do like the addition of `.fit`.  My only worry is that users might think that SciKeras always creates a `tf.data.Dataset`, which is not the case, by default it gives numpy arrays to `Keras.fit`. Do you think `Keras.fit(dataset or np.array)` makes that clear? It could also be dicts of lists or something, but that's at least more uncommon.

Reply:
What can `dataset_transformer` return? Only datasets/ndarrays? Or does it support the other arguments of Keras.fit?

Reply:
Anything that `Keras.fit` will accept. Internally, it looks something like this:

```python3
X, y, sample_weight = dataset_transformer.fit_transform((X, y, sample_weight))
model.fit(x=X, y=y, sample_weight=sample_weight)  # aka Keras.fit
```

Reply:
Maybe `Keras.fit(data)` is a good way to specify this? That way there's not confusion in interpreting `dataset` as `tf.dataDataset`. I can also add a small code block like in https://github.com/adriangb/scikeras/pull/167#discussion_r563994781 if that helps explain it.